firefoxThemes/options/options.js

```diff
@@ -1,51 +1,172 @@
+const onChangeSystemTheme = async (e) => {
+  browser.storage.local.get(["waifuThemes", "systemThemeChoice", "druthersThemes", "druthersBGs"])
+    .then(storage => {
+      for (const el of systemThemeRadios) {
+        initBox(el, el.id === e.target.id);
+      }
+      let isSystem = e.target.id === systemStates.SYSTEM
+        || e.target.id === systemStates.DRUTHERS;
+      if (isSystem) {
+        browser.browserSettings.overrideContentColorScheme.set({value: "system"});
+      } else {
+        browser.browserSettings.overrideContentColorScheme.set({value: "browser"});
+      }
+      // Show/Hide Druthers select options
+      displayDruthers(e.target.id, druthersSection);
+      browser.runtime.sendMessage({
+        resourceMSG: true,
+        isSystemRelated: true,
+        addObserver: isSystem
+      });
+      if (!storage.druthersBGs) {
+        storage.druthersBGs = new Map();
+        storage.druthersBGs.set(systemStates.LIGHT, backgroundTypes.PRIMARY);
+        storage.druthersBGs.set(systemStates.DARK, backgroundTypes.PRIMARY);
+      }
+      browser.storage.local.set({systemTheme: e.target.id, druthersBGs: storage.druthersBGs});
+    });
+}
+
+/*Change to a secondary light theme background*/
+function onChangeDruthersLightBG(e) {
+  onChangeDruthersBG(e, systemStates.LIGHT);
+}
+
+/*Change to a secondary dark theme background*/
+function onChangeDruthersDarkBG(e) {
+  onChangeDruthersBG(e, systemStates.DARK);
+}
+
+/*Change to a secondary background*/
+function onChangeDruthersBG(e, sysThemeType) {
+  browser.storage.local.get(["druthersThemes", "druthersBGs", "druthersThemes", "waifuThemes"])
+    .then(storage => {
+      e.target.className = !e.target.className ? 'checked' : '';
+    const backgroundType = e.target.className ? backgroundTypes.SECONDARY : backgroundTypes.PRIMARY;
+    storage.druthersBGs.set(sysThemeType, backgroundType);
+    initDruthersBG(storage.druthersBGs, systemStates.DRUTHERS, storage.druthersThemes, storage.waifuThemes.themes);
+    browser.storage.local.set({backgroundType, druthersBGs: storage.druthersBGs});
+    browser.runtime.sendMessage({
+      resourceMSG: true,
+      currentThemeId: storage.druthersThemes.get(sysThemeType)
+    });
+  });
 }
-/*Changes checkbox state*/
-function changeCheckboxState(e){
+
+/*Checks the boxes of an element to its opposite status*/
+function changeCheckedState(e) {
   const el = e.target;
-  el.className = el.className !== 'checked' ? "checked":'';
+  initBox(el, !!!el.className);
 }
 
 initContent();
-loadOnStartCheckbox.addEventListener('click', onChangeCheckEvents, true);
-textSelectionCheckbox.addEventListener('click', onChangeCheckEvents, true);
-scrollbarCheckbox.addEventListener('click', onChangeCheckEvents, true);
+clickListener(loadOnStartCheckbox, onChangeCheckEvents);
+clickListener(textSelectionCheckbox, onChangeCheckEvents);
+clickListener(scrollbarCheckbox, onChangeCheckEvents);
+multiClickListener(systemThemeRadios, onChangeSystemTheme);
```

Comment:
Do you think we could update the user experience for this?
Right now, if I select Druthers, it will set the theme to  the defaults, and it reloads the browser, then I have to go back to the options page again to make changes. (Same with changing anything in the Druther dropdowns)

Can we get a save/apply button that defers the changes until completed? That way the users don't have to keep going back to the settings.

Reply:
Yes! This is a better solution.